javaparser-core-generators/src/main/java/com/github/javaparser/generator/core/CoreGenerator.java

```diff
@@ -23,79 +23,170 @@
 
 import com.github.javaparser.ParserConfiguration;
 import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
 import com.github.javaparser.generator.core.node.*;
 import com.github.javaparser.generator.core.other.BndGenerator;
+import com.github.javaparser.generator.core.other.RemoveGeneratorAnnotations;
 import com.github.javaparser.generator.core.other.TokenKindGenerator;
 import com.github.javaparser.generator.core.visitor.*;
 import com.github.javaparser.utils.Log;
 import com.github.javaparser.utils.SourceRoot;
 
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 import static com.github.javaparser.ParserConfiguration.LanguageLevel.RAW;
+import static com.github.javaparser.utils.CodeGenerationUtils.f;
 
 /**
  * Generates all generated visitors in the javaparser-core module.
  * Suggested usage is by running the run_core_generators.sh script.
  * You may want to run_metamodel_generator.sh before that.
  */
 public class CoreGenerator {
+
+    public static final String GENERATOR_INIT_TIMESTAMP = System.currentTimeMillis() + "L";
+
     private static final ParserConfiguration parserConfiguration = new ParserConfiguration()
             .setLanguageLevel(RAW)
 //                                .setStoreTokens(false)
 //                                .setAttributeComments(false)
 //                                .setLexicalPreservationEnabled(true)
             ;
 
-    public static void main(String[] args) throws Exception {
+    protected final List<SourceRoot> sourceRoots;
+
+    private final SourceRoot jpCoreSourceRoot;
+    private final SourceRoot jssLogicSourceRoot;
+    private final SourceRoot jssModelSourceRoot;
+    private final SourceRoot jssSourceRoot;
+    private final SourceRoot generatedJavaCcSourceRoot;
+
+
+    public CoreGenerator(Path projectRoot) {
+        // Setup source roots
+        this.jpCoreSourceRoot = getSourceRootForJpModule(projectRoot, "javaparser-core");
+        this.jssLogicSourceRoot = getSourceRootForJpModule(projectRoot, "javaparser-symbol-solver-logic");
+        this.jssModelSourceRoot = getSourceRootForJpModule(projectRoot, "javaparser-symbol-solver-model");
+        this.jssSourceRoot = getSourceRootForJpModule(projectRoot, "javaparser-symbol-solver-core");
+
+        final Path generatedJavaCcRoot = projectRoot.resolve("javaparser-core").resolve("target").resolve("generated-sources").resolve("javacc");
+        this.generatedJavaCcSourceRoot = new SourceRoot(generatedJavaCcRoot, parserConfiguration);
+
+        // Setup collection for later iteration
+        this.sourceRoots = new ArrayList<>(7);
+        this.sourceRoots.add(this.jpCoreSourceRoot);
+        this.sourceRoots.add(this.jssLogicSourceRoot);
+        this.sourceRoots.add(this.jssModelSourceRoot);
```

Comment:
Are we generating anything in JSS? I'd prefer to start on a rewrite instead of putting more time in that project.

Reply:
Locally I do. 

I have annotated fields that should be used within an equals/hashcode method, and created a generator that creates those methods based on that annotation. This helps to keep them consistent.

Reply:
Is that something that could come into main repo? It sounds useful, although it is a little fuzzy what it does.

Reply:
That's the plan -- I'm just trying to figure out how to pass things up in bitesize logical chunks rather than en-masse :)